src/connection.ts

```diff
@@ -196,8 +196,18 @@ const DEFAULT_LANGUAGE = 'us_english';
  */
 const DEFAULT_DATEFORMAT = 'mdy';
 
-interface AzureActiveDirectoryMsiAppServiceAuthentication {
-  type: 'azure-active-directory-msi-app-service';
+export enum SupportedAuthenticationTypes {
+  AzureActiveDirectoryMsiAppService = 'azure-active-directory-msi-app-service',
+  AzureActiveDirectoryMsiVm = 'azure-active-directory-msi-vm',
+  AzureActiveDirectoryAccessToken = 'azure-active-directory-access-token',
+  AzureActiveDirectoryPassword = 'azure-active-directory-password',
+  AzureActiveDirectoryServicePrincipalSecret = 'azure-active-directory-service-principal-secret',
+  Ntlm = 'ntlm',
+  Default = 'default',
+}
```

Comment:
How do you feel about using an object here, instead of an enum? I find TypeScript enums to be weird - event the TypeScript docuementation suggests using objects instead:

```suggestion
export const SupportedAuthenticationTypes = {
  AzureActiveDirectoryMsiAppService: 'azure-active-directory-msi-app-service',
  AzureActiveDirectoryMsiVm: 'azure-active-directory-msi-vm',
  AzureActiveDirectoryAccessToken: 'azure-active-directory-access-token',
  AzureActiveDirectoryPassword: 'azure-active-directory-password',
  AzureActiveDirectoryServicePrincipalSecret: 'azure-active-directory-service-principal-secret',
  Ntlm: 'ntlm',
  Default: 'default',
} as const;
```

I'd also suggest changing the name to `AuthenticationTypes`, the fact that they're supported is sort of implicit. 😅

Reply:
Sure, but as we exporting them, we have to make sure that it won't be changed, so I'll make it as non-mutable by freezing that, like that:

```typescript
export const AuthenticationTypes  = Object.freeze({
  AzureActiveDirectoryMsiAppService = 'azure-active-directory-msi-app-service',
  AzureActiveDirectoryMsiVm = 'azure-active-directory-msi-vm',
  AzureActiveDirectoryAccessToken = 'azure-active-directory-access-token',
  AzureActiveDirectoryAccessTokenGenerator = 'azure-active-directory-access-token-generator',
  AzureActiveDirectoryPassword = 'azure-active-directory-password',
  AzureActiveDirectoryServicePrincipalSecret = 'azure-active-directory-service-principal-secret',
  Ntlm = 'ntlm',
  Default = 'default',
});
```

Reply:
**Update:**
We have to use enum, as because we won't able to use it as type in the interface. After the refactor it looks like: 
```typescript
export interface AzureActiveDirectoryMsiAppServiceAuthentication {
  type: SupportedAuthenticationTypes.AzureActiveDirectoryMsiAppService;
  options: {
    // ...
  };
}
```

When we change it as a const, it won't build.

Reply:
We don't really do this with any other objects we export, so I'm not really sure this needs to be done. If people want to shoot themselves in the foot, it's on them. 🤷 

But if you want to go down this path, you need to define this like this:

```
export const AuthenticationTypes = Object.freeze({
  AzureActiveDirectoryMsiAppService: 'azure-active-directory-msi-app-service',
  AzureActiveDirectoryMsiVm: 'azure-active-directory-msi-vm',
  AzureActiveDirectoryAccessToken: 'azure-active-directory-access-token',
  AzureActiveDirectoryAccessTokenGenerator: 'azure-active-directory-access-token-generator',
  AzureActiveDirectoryPassword: 'azure-active-directory-password',
  AzureActiveDirectoryServicePrincipalSecret: 'azure-active-directory-service-principal-secret',
  Ntlm: 'ntlm',
  Default: 'default',
} as const);
```

I still do think that what I suggested should be enough - a `... as const` definition is always marked as `Readonly`, so it shouldn't be accidentally changeable from typescript, without requiring the use of `freeze`. But either solution is fine.

Reply:
Readonly variables are understandable only for TypeScript when someone will use the `tedious` package without typescript, it would be allowed to modify the code, but as you told: 
> If people want to shoot themselves in the foot, it's on them.  

I totally agree with you 😄 

Backing to the option between const and enum, as I said, we have to use enum here.
As I tried to change it to the const, I couldn't build the project, because the following error occurred: 
> TS2503: Cannot find namespace 'AuthenticationTypes'.